lib/test_prof/event_prof/minitest.rb

```diff
@@ -1,3 +1,73 @@
 # frozen_string_literal: true
 
-# TODO: write Minitest reporter
+require 'test_prof/logging'
+require 'test_prof/event_prof/formatters/minitest'
+
+module Minitest
+  class EventProfReporter < AbstractReporter # :nodoc:
+    include TestProf::Logging
+
+    def initialize(options)
+      @profiler = configure_profiler(options)
+      @formatter = TestProf::EventProf::MinitestFormatter.new(@profiler)
+      @current_group = nil
+      @current_example = nil
+    end
+
+    def prerecord(group, example)
+      change_current_group(group, example) unless @current_group
+      track_current_example(group, example)
+    end
+
+    def record(*)
+      @profiler.example_finished(@current_example)
+    end
+
+    def report
+      @profiler.group_finished(@current_group)
+      result = @formatter.prepare_results
+
+      log :info, result
+    end
+
+    private
+
+    def track_current_example(group, example)
+      unless @current_group[:name] == group.name
+        @profiler.group_finished(@current_group)
+        change_current_group(group, example)
+      end
+
+      @current_example = {
+        name: example.gsub(/^test_\d+_/, ''),
+        location: File.expand_path(location(group, example).join(':')).gsub(Dir.getwd, '.')
+      }
+
+      @profiler.example_started(@current_example)
+    end
+
+    def change_current_group(group, example)
+      @current_group = {
+        name: group.name,
+        location: File.expand_path(location(group, example).first).gsub(Dir.getwd, '.')
+      }
+
+      @profiler.group_started(@current_group)
+    end
+
+    def location(group, example)
+      name = group.public_instance_methods(false).find { |mtd| mtd.to_s == example }
+      group.instance_method(name).source_location
```

Comment:
I've tried to run it with sample Rails app and got the following:

```
Run options: --event-prof=sql.active_record --seed 5773

# Running:

..../usr/local/lib/ruby/gems/2.4.0/bundler/gems/test-prof-3b646bac1b96/lib/test_prof/event_prof/minitest.rb:60:in `instance_method': nil is not a symbol nor a string (TypeError)
	from /usr/local/lib/ruby/gems/2.4.0/bundler/gems/test-prof-3b646bac1b96/lib/test_prof/event_prof/minitest.rb:60:in `location'
	from /usr/local/lib/ruby/gems/2.4.0/bundler/gems/test-prof-3b646bac1b96/lib/test_prof/event_prof/minitest.rb:52:in `change_current_group'
	from /usr/local/lib/ruby/gems/2.4.0/bundler/gems/test-prof-3b646bac1b96/lib/test_prof/event_prof/minitest.rb:38:in `track_current_example'
	from /usr/local/lib/ruby/gems/2.4.0/bundler/gems/test-prof-3b646bac1b96/lib/test_prof/event_prof/minitest.rb:19:in `prerecord'
```

I think, this is something Rails specific but we should care.

Example app is here https://github.com/palkan/th-dummy/tree/test-prof-minitest

Reply:
Also, FYI https://github.com/rails/rails/issues/29469 (maybe, it worth mentioning this issue in the guide).